src/js/services/ng-message.js

```diff
@@ -41,13 +41,11 @@ define('services/ng-message',[
                     };
                     ws.onmessage = function(msg) {
                         var data = JSON.parse(msg.data);
+                        var headers = JSON.parse(msg.headers);
                         var topic = data.topic;
 
-                        msg.from = data.data._token;
-                        msg.fromMe = data.data._token === token;
-                        delete data.data._token;
-                        if(Object.keys(data).length === 0)
-                            delete data.data;
+                        msg.from = headers["scoring-token"];;
+                        msg.fromMe = msg.from === token;
```

Comment:
Rather ugly to mutate `msg`. It smells a bit like bad design. I would expect `from` and `fromMe` to only be used inside this file. Possibly the listeners should indicate whether they are interested in hearing their own stuff back.

Alternatively, you could just do that extra scores load ;)

Reply:
I see what you're saying. It's more of a metadata header. Some listeners might want to receive both local and remote messages, but want to treat them differently. Also they might want to know who sent it. I wanted to expose that data to the listener to let him decide for itself what it would like to do with it. In our example, yes, it does decide no to listen to it. But I can think of different examples where that's not the case